src/list.c

```diff
@@ -68,3 +107,122 @@ int list_add(struct list_t *list, struct entry_t *entry) {
     previous_node->next = new_node;
     return 0;
 }
+
+/* Função que elimina da lista a entry com a chave key, libertando a
+ * memória ocupada pela entry.
+ * Retorna 0 se encontrou e removeu a entry, 1 se não encontrou a entry,
+ * ou -1 em caso de erro.
+ */
+int list_remove(struct list_t *list, char *key) {
+    // TODO update list size if element was removed
+
+    if (!list || !key) return -1;
+
+    if (!list->head) {
+        return 1;
+    }
+
+    int cmp_ret = strcmp(list->head->entry->key, key);
+    if (!cmp_ret) {
+        list->head = list->head->next;
+        return entry_destroy(list->head->entry);
```

Comment:
#If you entry_destroy the head after changing the head, wouldn't it delete the wrong entry?